org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService.xtend

```diff
@@ -9,59 +9,86 @@ package org.eclipse.xtext.ide.server.rename
 
 import com.google.inject.Inject
 import com.google.inject.Provider
+import org.eclipse.emf.ecore.EObject
 import org.eclipse.emf.ecore.util.EcoreUtil
+import org.eclipse.lsp4j.Position
 import org.eclipse.lsp4j.RenameParams
 import org.eclipse.lsp4j.WorkspaceEdit
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.eclipse.xtext.TerminalRule
+import org.eclipse.xtext.ide.refactoring.IRenameStrategy2
 import org.eclipse.xtext.ide.refactoring.RenameChange
 import org.eclipse.xtext.ide.refactoring.RenameContext
 import org.eclipse.xtext.ide.serializer.IChangeSerializer
+import org.eclipse.xtext.ide.server.Document
 import org.eclipse.xtext.ide.server.UriExtensions
 import org.eclipse.xtext.ide.server.WorkspaceManager
+import org.eclipse.xtext.nodemodel.ILeafNode
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils
+import org.eclipse.xtext.parsetree.reconstr.impl.TokenUtil
 import org.eclipse.xtext.resource.EObjectAtOffsetHelper
+import org.eclipse.xtext.resource.IResourceServiceProvider
 import org.eclipse.xtext.resource.XtextResource
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsProvider
 import org.eclipse.xtext.util.CancelIndicator
+
 import static org.eclipse.xtext.ide.refactoring.RefactoringIssueAcceptor.Severity.*
-import org.eclipse.xtext.ide.refactoring.IRenameStrategy2
-import org.eclipse.xtext.resource.impl.ResourceDescriptionsProvider
 
 /**
  * @author koehnlein - Initial contribution and API
  * @since 2.13
  */
-class RenameService implements IRenameService {
+@Accessors(PROTECTED_GETTER)
+class RenameService implements IRenameService, IRenameServiceExtension {
 	
-	@Inject extension EObjectAtOffsetHelper
+	@Inject extension EObjectAtOffsetHelper eObjectAtOffsetHelper
 	
 	@Inject IRenameStrategy2 renameStrategy 
 	
 	@Inject ChangeConverter.Factory converterFactory
 	
-	@Inject extension UriExtensions
+	@Inject extension UriExtensions uriExtensions
 	
 	@Inject Provider<IChangeSerializer> changeSerializerProvider
 	
 	@Inject Provider<ServerRefactoringIssueAcceptor> issueProvider
 	
+	@Inject IResourceServiceProvider.Registry serviceProviderRegistry
+	
+	@Inject TokenUtil tokenUtil
+	
+	/**
+	 * @deprecated use {@link #rename(WorkspaceManager, RenameParams, Options, CancelIndicator)} 
+	 *    instead.
+	 */
+	@Deprecated
 	override rename(WorkspaceManager workspaceManager, RenameParams renameParams, CancelIndicator cancelIndicator) {
+		rename(workspaceManager, renameParams, new Options(false), cancelIndicator)
+	}
+	
+	override rename(WorkspaceManager workspaceManager, RenameParams renameParams, Options options, CancelIndicator cancelIndicator) {
```

Comment:
I'm not sure if need the @since tag here too

Reply:
Fixed